contracts/scripts/deployments/diamonds/DeploySpaceFactory.s.sol

```diff
@@ -184,6 +191,10 @@ contract DeploySpaceFactory is DiamondHelper, Deployer {
     // legacy
     legacyArchitect = deployMockLegacyArchitect.deploy(deployer);
     spaceProxyInitializer = deploySpaceProxyInitializer.deploy(deployer);
+    spaceFactoryInit = deploySpaceFactoryInit.deploy(deployer);
+    spaceFactoryInitData = deploySpaceFactoryInit.makeInitData(
```

Comment:
Is the idea that if we needed to change spaceProxyInitializer or another proxy param after a diamond cut, we'd do so with an action ? Or how would we do so given this pr sets it once as init for every diamond deployment ?

Reply:
the idea is to have a way to set state during the upgrade, if done through an action we would deploy spaceFactoryInit and fill in the init field, we would also be able to set the spaceFactoryInitData in the initPayload field

![Screenshot 2024-09-12 at 8 12 00 PM](https://github.com/user-attachments/assets/ed8ec452-5827-4ad8-8632-4674ae350169)

Since we're not deploying new diamonds anymore and only facets, the current DeploySpaceFactory and other diamond deployment scripts are limited by not being able to set new state. This setup allows for that and works for both alpha, gamma and omega deployments.